### DISCUSSION TITLE
Generating types for Dart

### DISCUSSION THREAD

**Discussion by randreu28** [supabase/supabase#12530] [Feature Requests] [answered]
There's an official way [mentioned in the docs](https://supabase.com/docs/guides/database/api/generating-types) to create TypeScript types automatically with a CLI, and even creating a github CI for it.

I wish there would be something similar for the Dart Language. Dart is a very typed language and missing the types when getting the data from your db is crucial and a very good feature for the future.

**Comment by VanesaSolas4N** [comment]
It's true that TypeScript provides a CLI tool for generating types automatically, which can be very helpful for maintaining type safety and reducing errors in large codebases. However, I'm not aware of a similar tool for Dart at the moment.

That being said, there are some libraries and tools available for Dart that can help with type generation and code analysis, such as build_runner and analyzer. Additionally, the json_serializable package provides a convenient way to generate Dart classes from JSON data, which can help ensure type safety when working with APIs or databases.

Overall, while there may not be an official CLI tool for generating types in Dart, there are still options available for ensuring type safety and reducing errors in your code.

**Comment by sunilguptasg** [comment]
I'm just looking for static analysis - help during development, to ensure no mistakes in spelling etc. Not looking for JSON<>Dart. For performance I often directly work with the JSON and do not always go through a Dart class (irrespective of this being considered bad practice).

**Comment by mmvergara** [comment, marked as answer]
Hello, i made a [tool for generating dart classes from your Supabase schema](https://github.com/mmvergara/supabase-schema-dart-class-generator). I don't know if you guys still need it but here it is i guess

```dart
// fetchedBooks is a typeof List<Books>
final books = await supabase
      .books
      .select("*")
      .withConverter((data) => data.map(Books.fromJson).toList());
```
supports other cruds
